staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -106,7 +106,18 @@ import (
 // and index updates happen before such a prescribed notification.
 // For a given SharedInformer and client, the notifications are
 // delivered sequentially.  For a given SharedInformer, client, and
-// object ID, the notifications are delivered in order.
+// object ID, the notifications are delivered in order.  Because
+// `ObjectMeta.UID` has no role in identifying objects, if an object
+// O1 with ID (e.g. namespace and name) X and `ObjectMeta.UID` U1 in
+// the SharedInformer's local cache is deleted and shortly after
```

Comment:
We have no meaningful characterization of "shortly" here, so I would drop it.  I would replace "shortly after" with "later.  But this is just a suggestion.